### PR TITLE
[Docs] update `rowNumberInXYZ` functions

### LINKED ISSUE #2007: Document `rowNumberInXYZ` functions
Create documentation for the `rowNumberInXYZ` function. See #1833 for details on how to complete this issue.

- [ ] rowNumberInAllBlocks
- [ ] rowNumberInBlock

Child issue of #1833.

### PR DESCRIPTION
**Pull request by Blargian**
Closes [#2007](https://github.com/ClickHouse/clickhouse-docs/issues/2007) as part of the [functions project](https://github.com/ClickHouse/clickhouse-docs/issues/1833).

Updates documentation for:
- `blockNumber`
- `rowNumberInBlock`
- `rowNumberInAllBlocks`

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 4483da06e819e902c273d6e84d3d58971d73c30c with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63849/4483da06e819e902c273d6e84d3d58971d73c30c/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: docs/en/sql-reference/functions/other-functions.md [hunk 1]
```diff
@@ -998,17 +998,110 @@ SELECT version()
...
+
+```response
+┌─min(rowNumberInBlock())─┬─max(rowNumberInBlock())─┐
+│                       0 │                   65408 │
+└─────────────────────────┴─────────────────────────┘
+```
+
+## rowNumberInAllBlocks
+
+Returns the ordinal number of the row in the [data block](https://clickhouse.com/docs/en/integrations/python#data-blocks). This function only considers the affected data blocks.
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/other-functions.md:1072]: `Returns a unique row number for each row processed by `rowNumberInAllBlocks`. The returned numbers start at 0.`

File: docs/en/sql-reference/functions/other-functions.md [hunk 2]
```diff
@@ -998,17 +998,110 @@ SELECT version()
...
+
+**Returned value**
+
+- Ordinal number of the row in the data block starting from 0. [UInt64](../data-types/int-uint.md).
+
+**Example**
+
+Query:
+
+```sql
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/other-functions.md:1088]: Not a useful example, let's do

```sql
SELECT rowNumberInAllBlocks()
FROM
(
    SELECT *
    FROM system.numbers_mt
    LIMIT 10
)
SETTINGS max_block_size = 2
```

File: docs/en/sql-reference/functions/other-functions.md [hunk 3]
```diff
@@ -998,17 +998,110 @@ SELECT version()
...
+SELECT
+    min(rowNumberInBlock()),
+    max(rowNumberInBlock())
+FROM (SELECT * FROM system.numbers_mt LIMIT 100000);
+```
+
+Result:
+
+```response
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/other-functions.md:1064]: Not a useful example, let's do

```sql
Not a useful example, let's do

```sql
SELECT rowNumberInBlock()
FROM
(
    SELECT *
    FROM system.numbers_mt
    LIMIT 10
)
SETTINGS max_block_size = 2
```

**Review by rschu1ze — CHANGES_REQUESTED** [review]

**Review by rschu1ze — APPROVED** [review]

**Pull request merged**